dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/ClientCore.cs

```diff
@@ -717,10 +719,16 @@ static void AddResponseMessage(ChatCompletionsOptions chatOptions, ChatHistory c
                         streamedContents?.Add(openAIStreamingChatMessageContent);
                         yield return openAIStreamingChatMessageContent;
                     }
+
+                    // Translate all entries into ChatCompletionsFunctionToolCall instances.
+                    toolCalls = OpenAIFunctionToolCall.ConvertToolCallUpdatesToChatCompletionsFunctionToolCalls(
+                        ref toolCallIdsByIndex, ref functionNamesByIndex, ref functionArgumentBuildersByIndex);
+                    // Translate all entries into FunctionCallContent instances for diagnostics purposes.
+                    functionCallContents = activity is null ? null : toolCalls.Select(this.GetFunctionCallContent).ToArray();
```

Comment:
Rather than doing all this translation, can we just get the original response from the Azure SDK object? Doesn't it make it available as a raw response content off of the relevant objects? It'd be both cheaper and higher fidelity.

Reply:
Is checking `activity is null` the right check here, or do we need to check if the sensitive model diagnostics are enabled? It'd be good to avoid doing such work except in the case where it's actually going to be used.

Reply:
The reason I didn't do that is because I would have to do the parsing inside the model diagnostic module, which is supposed to be agnostic to the model specifics.

The `FunctionCallContent` class carries enough information to represent the tool calls and is a common class. It's used in the non-streaming API, and AFAIK the Gemini connector is going to adapt to it soon.

Reply:
> Is checking `activity is null` the right check here, or do we need to check if the sensitive model diagnostics are enabled? It'd be good to avoid doing such work except in the case where it's actually going to be used.

Nice catch! I will address this